### PR TITLE
[giflib] Fix multi-config `find_package(GIF)`

### LINKED ISSUE #26833: [giflib] Debug library linked in release configuration
**Host Environment**

- OS: Windows 10 & 11
- Compiler: Visual Studio 2015, 2019, 2022

**To Reproduce**

CMake project + vcpkg manifest.

vcpkg.json:
```json
{
    "$schema": "https://raw.githubusercontent.com/microsoft/vcpkg/master/scripts/vcpkg.schema.json",
    "name": "vcpkg-giflib-bug",
    "version": "1.0.0",
    "builtin-baseline": "a1737683a6f8ec5d8673890581c2a8c55838d411",
    "dependencies": [ "giflib" ]
}
```
CMakeLists.txt:
```cmake
cmake_minimum_required(VERSION 3.16)
project(vcpkg-giflib-bug)
find_package(GIF REQUIRED)
add_executable(vcpkg-giflib-bug main.cpp)
target_link_libraries(vcpkg-giflib-bug PRIVATE GIF::GIF)
```
main.cpp:
```c++
void main() { }
```

image

All other libraries that I used except Giflib were linked correctly.

### PR DESCRIPTION
**Pull request by dg0yt**
- #### What does your PR fix?
  Fixes https://github.com/microsoft/vcpkg/issues/26833.
  Backports official `GIF::GIF` target for older versions of CMake.
  Alternative to #26869 which takes wrong direction IMO. CC @LilyWangLL 

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  unchanged, no

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  yes

### CODE REVIEW

File: ports/giflib/vcpkg-cmake-wrapper.cmake [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    )
+    if("${GIF_LIBRARY_DEBUG}")
+        set_property(TARGET GIF::GIF APPEND PROPERTY
+            IMPORTED_CONFIGURATIONS DEBUG)
+        set_target_properties(GIF::GIF PROPERTIES
+            IMPORTED_LINK_INTERFACE_LANGUAGES_DEBUG "C"
+            IMPORTED_LOCATION_DEBUG "${GIF_LIBRARY_DEBUG}")
+    endif()
+endif()
+_find_package(${ARGS})
```
Comment (Neumann-A) [inline ports/giflib/vcpkg-cmake-wrapper.cmake:25]:
```suggestion
_find_package(${ARGS})
if(NOT TARGET GIF::GIF)
    add_library(GIF::GIF UNKNOWN IMPORTED)
    set_target_properties(GIF::GIF PROPERTIES
        INTERFACE_INCLUDE_DIRECTORIES "${GIF_INCLUDE_DIR}"
        IMPORTED_CONFIGURATIONS RELEASE
        IMPORTED_LINK_INTERFACE_LANGUAGES_RELEASE "C"
        IMPORTED_LOCATION_RELEASE "${GIF_LIBRARY_RELEASE}"
    )
    if("${GIF_LIBRARY_DEBUG}")
        set_property(TARGET GIF::GIF APPEND PROPERTY
            IMPORTED_CONFIGURATIONS DEBUG)
        set_target_properties(GIF::GIF PROPERTIES
            IMPORTED_LINK_INTERFACE_LANGUAGES_DEBUG "C"
            IMPORTED_LOCATION_DEBUG "${GIF_LIBRARY_DEBUG}")
    endif()
endif()
```
order ?

**Comment by LilyWangLL** [comment]
Usage test error:
```
1> [CMake] CMake Error at E:/Lily/giflib/installed/x64-windows/share/gif/vcpkg-cmake-wrapper.cmake:26 (_find_package):
1> [CMake]   Could not find a package configuration file provided by "gif" with any of
1> [CMake]   the following names:
1> [CMake] 
1> [CMake]     gifConfig.cmake
1> [CMake]     gif-config.cmake
1> [CMake] 
1> [CMake]   Add the installation prefix of "gif" to CMAKE_PREFIX_PATH or set "gif_DIR"
1> [CMake]   to a directory containing one of the above files.  If "gif" provides a
1> [CMake]   separate development package or SDK, be sure it has been installed.

```

**Comment by dg0yt** [comment]
> Usage test error:
> 
> ```
> 1> [CMake] CMake Error at E:/Lily/giflib/installed/x64-windows/share/gif/vcpkg-cmake-wrapper.cmake:26 (_find_package):
> 1> [CMake]   Could not find a package configuration file provided by "gif" with any of
> 1> [CMake]   the following names:
> 1> [CMake] 
> 1> [CMake]     gifConfig.cmake
> 1> [CMake]     gif-config.cmake
> 1> [CMake] 
> 1> [CMake]   Add the installation prefix of "gif" to CMAKE_PREFIX_PATH or set "gif_DIR"
> 1> [CMake]   to a directory containing one of the above files.  If "gif" provides a
> 1> [CMake]   separate development package or SDK, be sure it has been installed.
> ```

## USER ERROR, NOT USAGE ERROR 😠

**Comment by Neumann-A** [comment]
@LilyWangLL: It is `find_package(GIF)` not `find_package(gif)` ... case is important with find modules.

**Comment by dg0yt** [comment]
Well, it is written in https://github.com/microsoft/vcpkg/issues/26833#issue-1376082385.

**Comment by LilyWangLL** [comment]
> @LilyWangLL: It is `find_package(GIF)` not `find_package(gif)` ... case is important with find modules.

I know, but I test this PR locally, after installing `giflib`, the output is:
```
Elapsed time to handle giflib:x64-windows: 10.37 s
Total elapsed time: 17.63 s

giflib provides CMake targets:

    # this is heuristically generated, and may not be correct
    find_package(gif CONFIG REQUIRED)
    target_link_libraries(main PRIVATE GIF::GIF)
```

**Comment by LilyWangLL** [comment]
I just clone the https://github.com/dg0yt/vcpkg.git and switch to branch dg0yt:giflib.

**Comment by LilyWangLL** [comment]
And I also test the usage in a new CMake project:
```
find_package(GIF REQUIRED)
target_link_libraries(main PRIVATE GIF::GIF)
```
But the issue #26833 doesn't resolved, the link library still is debug lib.
image
Please let me know if there is a problem with my operation

**Comment by Neumann-A** [comment]
> Please let me know if there is a problem with my operation

Manifest mode?

**Comment by dg0yt** [comment]
> ~~~
> giflib provides CMake targets:
> 
>     # this is heuristically generated, and may not be correct
>     find_package(gif CONFIG REQUIRED)
>     target_link_libraries(main PRIVATE GIF::GIF)

https://github.com/microsoft/vcpkg/issues/20190. There, I was increasing the size of the 

> # Bad User Experience

headline until I run out of levels.

Would be fixed by https://github.com/microsoft/vcpkg/issues/20190#issuecomment-1072067545.

For now I can add a usage file to workaround the vpckg tool UX bug.

**Comment by LilyWangLL** [comment]
> > Please let me know if there is a problem with my operation
> 
> Manifest mode?

No, I test it in simple CMake project by integrate vcpkg.

**Comment by dg0yt** [comment]
Done.

**Comment by LilyWangLL** [comment]
> And I also test the following usage in a new CMake project:
> 
> ```
> find_package(GIF REQUIRED)
> target_link_libraries(main PRIVATE GIF::GIF)
> ```
> 
> But the issue #26833 doesn't resolved, the link library still is debug lib. image Please let me know if there is a problem with my operation

This issue doesn't resolve.

**Comment by dg0yt** [comment]
FTR this is the output from the multi-config testing in #27369:
~~~
-- Found GIF: optimized;D:/installed/x64-windows/lib/gif.lib;debug;D:/installed/x64-windows/debug/lib/gif.lib (found version "5.2.1") 
~~~

**Comment by LilyWangLL** [comment]
I removed all caches and test again, issue #26833 resolved.

**Review by LilyWangLL — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the fix!

**Pull request merged**